File: frontend/src/types/booking.ts
export type BookingStatus = "PENDING" | "CONFIRMED" | "COMPLETED" | "CANCELLED";
export type PaymentStatus = "UNPAID" | "PARTIAL" | "PAID" | "REFUNDED";

export interface Booking {
  id: string;
  bookingRef: string;
  userId: string;
  safari?: {
    id: string;
    name: string;
    slug: string;
  };
  vehicle?: {
    id: string;
    name: string;
    slug: string;
  };
  destination?: {
    id: string;
    name: string;
    slug: string;
  };
  startDate: string;
  endDate: string;
  participants: number;
  totalPrice: number;
  currency: string;
  status: BookingStatus;
  paymentStatus: PaymentStatus;
  customerName: string;
  customerEmail: string;
  customerPhone?: string;
  specialRequests?: string;
  adminNotes?: string;
  createdAt: string;
}

export interface CreateBookingRequest {
  safariId?: string;
  vehicleId?: string;
  destinationId?: string;
  startDate: string;
  endDate: string;
  participants: number;
  customerName: string;
  customerEmail: string;
  customerPhone?: string;
  specialRequests?: string;
}

File: backend/src/main/java/com/trek/rwanda/booking/Booking.java
package com.trek.rwanda.booking;

import com.trek.rwanda.destination.Destination;
import com.trek.rwanda.safari.Safari;
import com.trek.rwanda.user.User;
import com.trek.rwanda.vehicle.Vehicle;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "bookings")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Booking {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, unique = true, length = 20)
    private String bookingRef;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "safari_id")
    private Safari safari;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "vehicle_id")
    private Vehicle vehicle;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "destination_id")
    private Destination destination;

    @Column(nullable = false)
    private LocalDate startDate;

    @Column(nullable = false)
    private LocalDate endDate;

    @Column(nullable = false)
    @Builder.Default
    private Integer participants = 1;

    @Column(precision = 12, scale = 2)
    private BigDecimal totalPrice;

    @Builder.Default
    private String currency = "USD";

    @Column(nullable = false, length = 20)
    @Builder.Default
    private String status = "PENDING";

    @Column(nullable = false, length = 20)
    @Builder.Default
    private String paymentStatus = "UNPAID";

    @Column(nullable = false, length = 100)
    private String customerName;

    @Column(nullable = false, length = 255)
    private String customerEmail;

    @Column(length = 20)
    private String customerPhone;

    @Column(columnDefinition = "TEXT")
    private String specialRequests;

    @Column(columnDefinition = "TEXT")
    private String adminNotes;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
}
